symengine/sets.h

```diff
@@ -326,6 +367,23 @@ inline RCP<const Set> interval(const RCP<const Number> &start,
 }
 
 // ! \return RCP<const Set>
+inline RCP<const Set> imageset(const RCP<const Basic> &sym,
+                               const RCP<const Basic> &expr,
+                               const RCP<const Set> &base)
+{
+    if (not is_a<Symbol>(*sym))
+        throw SymEngineException("first arg is expected to be a symbol");
+
+    if (eq(*expr, *sym) or eq(*base, *emptyset()))
+        return base;
+
+    if (is_a_Number(*expr))
+        return finiteset({expr});
```

Comment:
We should follow SymPy here:
![image](https://user-images.githubusercontent.com/9252491/27456617-095a438a-57a2-11e7-90e7-eb22ab83d502.png)